tree/tree/src/TBasket.cxx

```diff
@@ -204,12 +178,51 @@ Int_t TBasket::DropBuffers()
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Calculates the entry offset array, if possible.
+///
+/// Result is cached, meaning that this should only be invoked once per basket.
+
+Int_t *TBasket::GetCalculatedEntryOffset()
+{
+   if (fEntryOffset != reinterpret_cast<Int_t *>(-1)) {
+      return fEntryOffset;
+   }
+
+   if (R__unlikely(!fBranch)) {
+      Error("GetCalculatedEntryOffset", "Basket entry offset calculation requested, but no associated TBranch!");
+      return nullptr;
+   }
+   if (R__unlikely(fBranch->GetNleaves() != 1)) {
+      Error("GetCalculatedEntryOffset", "Branch contains multiple leaves - unable to calculated entry offsets!");
+      return nullptr;
+   }
+   TLeaf *leaf = static_cast<TLeaf *>((*fBranch->GetListOfLeaves())[0]);
+   fEntryOffset = leaf->GenerateOffsetArray(fKeylen, fNevBuf);
+   return fEntryOffset;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Determine whether we can generate the offset array when this branch is read.
+///
+
+Bool_t TBasket::CanGenerateOffsetArray()
+{
+   if (fBranch->GetNleaves() != 1) {
+      return kFALSE;
+   }
+   TLeaf *leaf = static_cast<TLeaf *>((*fBranch->GetListOfLeaves())[0]);
+   return leaf->CanGenerateOffsetArray();
+}
+
+////////////////////////////////////////////////////////////////////////////////
 /// Get pointer to buffer for internal entry.
 
 Int_t TBasket::GetEntryPointer(Int_t entry)
 {
    Int_t offset;
-   if (fEntryOffset) offset = fEntryOffset[entry];
+   Int_t *entryOffset = GetEntryOffset();
+   if (entryOffset)
```

Comment:
If you change the if from one to two line, you must also change the else from one to two lines.

Reply:
I think this was `clang-tidy` nonsense.  Fixing.